functions/snippets/src/main/java/Tips.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_tips_scopes]
+// [START run_tips_global_scope]
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class Tips {
+  // Global (instance-wide) scope
+  // This computation runs at instance cold-start
+  public static final int INSTANCE_VAR = heavyComputation();
+
+  public void ScopesDemo(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, InterruptedException {
+    // Per-function scope
+    // This computation runs every time this function is called
+    final int functionVar = lightComputation();
+    PrintWriter writer = response.getWriter();
+    writer.write(String.format("Per instance: %d, per function: %d", INSTANCE_VAR, functionVar));
+  }
+  // [END functions_tips_scopes]
+  // [END run_tips_global_scope]
+
+  private static int lightComputation() {
```

Comment:
These functions are still excluded, which means that this snippet isn't copy-paste-runnable (see the [Sample Format Guidelines](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#specific-goals)). Older samples might not be following the format, but we are trying to keep all new samples following the guidelines. 

Also, both functions are exactly the same, which makes it a bit confusing to understand when reading. Perhaps it would be better to replace this with something like this:

```java
   // This function represents a heavy computation, which is only called when the instance starts.
   private static int heavyComputation() {
    TimeUnit.SECONDS.sleep(3);
    return System.currentTimeMillis();
  }

   // This function represents a light computation, which might be called on every incoming request.
   private static int lightComputation() {
    TimeUnit.SECONDS.sleep(0);
    return System.currentTimeMillis();
  }
```

Reply:
- These functions are excluded like they are in the Node sample.
- These functions aren't exactly the same. They are the same as Node.

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/4b2629e759eba9cf7d89aed3302bcf2795e1eecf/functions/tips/index.js#L18

What would you like us to do?

Reply:
> * These functions are excluded like they are in the Node sample.

I believe that the node repo has a similar policy regarding copy-paste-runnable (since it's borrowed from the overarching [Code Snippets Rubric](https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/samples/snippets.md?cl=head)). It looks like these samples probably predate this policy, but I don't know how the node repo is enforcing those rules.  

Currently, the java-docs-samples policy is that new samples should follow the guidelines, while minor updates to existing samples can be grandfathered in. Since these samples are new, they need to follow the guidelines and include these functions so that users can copy them, paste them, and run them without having to fill out these functions. 

> * These functions are not exactly the same. 

This is my bad, the version I was looking at shows them [both using addition](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/70e8f4319a05ecb77c900650089ac834f62ccbc4/functions/snippets/src/main/java/Tips.java#L41-L49), but I can see at HEAD they are different. 

At the very least, this version of the sample should contain the comments I posted above. (My two cents is still that the intent of the sample is significantly more clear when using sleep, because the user can see that the difference between running those functions is non-trivial. However if you strongly prefer to remain consistent with the Node version, I'm satisfied with the comments clarifying above)